Source/Core/VideoCommon/VertexLoader.cpp

```diff
@@ -312,38 +306,42 @@ void VertexLoader::CompileVertexTranslator()
   }
 
   // Texture matrix indices (remove if corresponding texture coordinate isn't enabled)
-  for (int i = 0; i < 8; i++)
+  for (size_t i = 0; i < m_VtxDesc.high.TexCoord.Size(); i++)
   {
     m_native_vtx_decl.texcoords[i].offset = nat_offset;
     m_native_vtx_decl.texcoords[i].type = VAR_FLOAT;
     m_native_vtx_decl.texcoords[i].integer = false;
 
-    const int format = m_VtxAttr.texCoord[i].Format;
-    const int elements = m_VtxAttr.texCoord[i].Elements;
+    const auto format = m_VtxAttr.texCoord[i].Format;
+    const auto elements = m_VtxAttr.texCoord[i].Elements;
 
-    if (tc[i] != NOT_PRESENT)
+    if (m_VtxDesc.high.TexCoord[i] != VertexComponentFormat::NotPresent)
```

Comment:
This code is just a bit more complicated looking because `m_VtxDesc.high.TexCoord[i]` is repeated a bit; I'll extract it into a variable named `tc` like `format` and `elements` above.